### PR TITLE
build: set up semantic-release, as well as a release workflow

### PR DESCRIPTION
**Pull request by patrickocoffeyo**
If the code looks good, merge. If it doesn't work, I'll fix it :D

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1,2 +1 @@
-# If you need to update this, please also update the version used in the `.github/workflows` files
```
Comment (patrickocoffeyo) [inline .nvmrc:1]: Had to remove this because it makes the file un-parseable to nvm when running `nvm use`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        with:
+          fetch-depth: 0
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.10
+      - name: Install
+        run: npm install
+      - name: Release
+        run: npm run semantic-release
```
Comment (ccjjmartin) [inline .github/workflows/release.yml:20]: @patrickocoffeyo I think `semantic-release` by itself normally adds a v in front of the version number so I normally pass the `--tag-format '${version}'` flag to get rid of the v and just keep the version number.  This is more consistent with how the PHP world does version numbers.  I know this is not a strictly PHP repo though, so maybe not applicable.

File: .nvmrc (hunk 1 above)
Comment (ccjjmartin) [reply .nvmrc:1]: I had to do the same on SOM recently

Comment (ModulesUnraveled) [reply .nvmrc:1]: Oh dang! ... do either of you know a better way to document "all the places node versions should match?"

**Review by ModulesUnraveled — COMMENTED** [review]
Everything looks good to me. I'll let @ccjjmartin give the official approval after you guys decide on the tag format. I don't think I care either way.

File: .github/workflows/release.yml (hunk 2 above)
Comment (ModulesUnraveled) [reply .github/workflows/release.yml]: This works for me. I couldn't find any reason to use one or the other.

**Review by ModulesUnraveled — APPROVED** [review]
Merging to test

**Pull request merged**